e2e/test/netstandard20/ProvisioningE2ETests.cs

```diff
@@ -584,7 +584,7 @@ public async Task ProvisioningDeviceClient_InvalidRegistrationId_TpmRegister_Htt
             }
             catch (ProvisioningTransportException ex)
             {
-                Assert.IsTrue(ex.Message.Contains("404201"));
+                Assert.IsTrue(ex.Message.Contains("HTTP transport exception"));
```

Comment:
do you have an example of this exception handy? this seems a bit odd that we have to rely on an exception message text ...

Reply:
+1 it would be good if there were a specific Exception code and http status combination to assert against.

Reply:
I wish there is a better way to check this but there's not any code I can assert for.

Reply:
what changed here, wrt behavior, that we cannot assert with 404201 anymore?

Reply:
apparently the message does not contain that string anymore (based on a screenshot i got from Binal) that's another reason why we shouldn't look for certain strings in an exception message since that is prone to change. but it doesn't seem like there is a status code on that exception type either. Who owns this exception? the Service? Can we ask for status code?

Reply:
So on this note, that behavior should not change as Abhipsa mentioned. I made sure to add a relevant catch block

Yes you are right @azabbasi , this is the exception we are converting too for the response. I made sure in new changes, it does capture this error code. I just added a note to better explain it.